packages/client/hmi-client/src/types/common.ts

```diff
@@ -188,7 +188,16 @@ export enum ChartSettingType {
 	INTERVENTION = 'intervention'
 }
 
-export interface ChartSetting {
+export type ChartSetting = ChartSettingBase | ChartSettingEnsembleVariable;
+
+export interface ChartSettingEnsembleVariable extends ChartSettingBase {
+	type: ChartSettingType.VARIABLE_ENSEMBLE;
```

Comment:
I dont think we should force this type for `ChartSettingEnsembleVariable`?
We may want other valid types for ensemble such as weights?

Reply:
I think you are free to add new type extending the base setting when needed. I just added this type specific for `ensemble variables over time` charts.